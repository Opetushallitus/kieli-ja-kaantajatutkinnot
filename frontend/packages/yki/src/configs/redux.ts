import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

import { rootReducer, setupStore } from 'redux/store/index';

export type RootState = ReturnType<typeof rootReducer>;
export type AppStore = ReturnType<typeof setupStore>;
type AppDispatch = AppStore['dispatch'];

export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
