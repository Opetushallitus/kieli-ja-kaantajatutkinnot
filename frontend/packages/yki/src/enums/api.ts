export enum APIEndpoints {
  ExamSessions = '/yki/api/exam-session',
  ExamSession = '/yki/api/exam-session/:examSessionId',
  Evaluations = '/yki/api/evaluation',
  Evaluation = '/yki/api/evaluation/:evaluationId',
  EvaluationOrder = '/yki/api/evaluation/:evaluationId/order',
  InitRegistration = '/yki/api/registration/init',
  LoginLink = '/yki/api/login-link',
  SubmitRegistration = '/yki/api/registration/:registrationId/submit',
  SuomiFiAuthRedirect = '/yki/auth/',
}
