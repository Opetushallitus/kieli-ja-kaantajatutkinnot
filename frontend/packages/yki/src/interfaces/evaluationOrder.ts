export interface ExaminationParts {
  readingComprehension: boolean;
  speechComprehension: boolean;
  speaking: boolean;
  writing: boolean;
}

export interface ParticipantDetails {
  firstNames?: string;
  lastName?: string;
  email?: string;
  birthdate?: string;
}

export type Subtest = 'LISTENING' | 'READING' | 'SPEAKING' | 'WRITING';

export interface EvaluationOrderRequest {
  first_names: string;
  last_name: string;
  email: string;
  birthdate: string;
  subtests: Array<Subtest>;
}

export interface EvaluationOrderResponse {
  redirect: string;
}
