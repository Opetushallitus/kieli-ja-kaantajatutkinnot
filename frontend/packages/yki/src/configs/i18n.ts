import i18n, { changeLanguage, t, use } from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import {
  initReactI18next,
  useTranslation,
  UseTranslationOptions,
} from 'react-i18next';
import { AppLanguage, I18nNamespace } from 'shared/enums';
import { DateUtils } from 'shared/utils';

import commonEN from 'public/i18n/en-GB/common.json';
import publicEN from 'public/i18n/en-GB/public.json';
import commonFI from 'public/i18n/fi-FI/common.json';
import publicFI from 'public/i18n/fi-FI/public.json';
import commonSV from 'public/i18n/sv-SE/common.json';
import publicSV from 'public/i18n/sv-SE/public.json';

// Defaults and resources
const langFI = AppLanguage.Finnish;
const langSV = AppLanguage.Swedish;
const langEN = AppLanguage.English;

const supportedLangs = [langFI, langSV, langEN];

const resources = {
  [langFI]: {
    [I18nNamespace.Common]: commonFI,
    [I18nNamespace.Public]: publicFI,
  },
  [langSV]: {
    [I18nNamespace.Common]: commonSV,
    [I18nNamespace.Public]: publicSV,
  },
  [langEN]: {
    [I18nNamespace.Common]: commonEN,
    [I18nNamespace.Public]: publicEN,
  },
};

const detectionOptions = {
  order: ['localStorage', 'htmlTag'],
  caches: ['localStorage'],
  lookupLocalStorage: 'i18NextLng_yki',
};

// TypeScript definitions for react-i18next. IDE might show this to be unused, but ts does some type checking with it.
declare module 'react-i18next' {
  interface CustomTypeOptions {
    defaultNS: typeof langFI;
    resources: {
      [langFI]: typeof commonFI;
      [langSV]: typeof commonSV;
      [langEN]: typeof commonEN;
    };
  }
}

export const initI18n = () => {
  const i18n = use(initReactI18next).use(LanguageDetector).init({
    defaultNS: 'common',
    resources,
    detection: detectionOptions,
    fallbackLng: langFI,
    load: 'currentOnly',
    debug: !REACT_ENV_PRODUCTION,
  });
  const currentLanguage = getCurrentLang() as AppLanguage;
  DateUtils.setDayjsLocale(currentLanguage);

  return i18n;
};

const useAppTranslation = (
  options: UseTranslationOptions<string>,
  ns: I18nNamespace
) => {
  // @ts-expect-error ts import fail
  return useTranslation(ns, options);
};

export const usePublicTranslation = (
  options: UseTranslationOptions<string>
) => {
  return useAppTranslation(options, I18nNamespace.Public);
};

export const useCommonTranslation = () => {
  const { t } = useAppTranslation(
    {
      keyPrefix: 'yki.common',
    },
    I18nNamespace.Common
  );

  return t;
};

// ts-unused-exports:disable-next-line
export const translateOutsideComponent = () => {
  return t;
};

export const getCurrentLang = (): AppLanguage => {
  return i18n.language as AppLanguage;
};

export const getSupportedLangs = (): Array<AppLanguage> => {
  return supportedLangs;
};

export const changeLang = (language: AppLanguage) => {
  DateUtils.setDayjsLocale(language);

  return changeLanguage(language);
};
