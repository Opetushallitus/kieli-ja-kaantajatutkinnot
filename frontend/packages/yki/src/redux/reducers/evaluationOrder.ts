import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { APIResponseStatus } from 'shared/enums';

import {
  EvaluationOrderDetails,
  ExaminationParts,
  ParticipantDetails,
} from 'interfaces/evaluationOrder';
import { EvaluationPeriod } from 'interfaces/evaluationPeriod';

export interface EvaluationOrderState {
  submitOrderState: APIResponseStatus;
  loadPeriodState: APIResponseStatus;
  loadEvaluationOrderDetailsState: APIResponseStatus;
  evaluationPeriod?: EvaluationPeriod;
  evaluationOrderDetails?: EvaluationOrderDetails;
  examinationParts: ExaminationParts;
  acceptConditions: boolean;
  participantDetails: ParticipantDetails;
  showErrors: boolean;
  evaluationPaymentRedirect?: string;
}

export const initialState: EvaluationOrderState = {
  submitOrderState: APIResponseStatus.NotStarted,
  loadPeriodState: APIResponseStatus.NotStarted,
  loadEvaluationOrderDetailsState: APIResponseStatus.NotStarted,
  examinationParts: {
    readingComprehension: false,
    speechComprehension: false,
    speaking: false,
    writing: false,
  },
  acceptConditions: false,
  participantDetails: {},
  showErrors: false,
};

const evaluationOrderSlice = createSlice({
  name: 'evaluationOrder',
  initialState,
  reducers: {
    loadEvaluationPeriod(state, _action: PayloadAction<number>) {
      state.loadPeriodState = APIResponseStatus.InProgress;
    },
    storeEvaluationPeriod(state, action: PayloadAction<EvaluationPeriod>) {
      state.loadPeriodState = APIResponseStatus.Success;
      state.evaluationPeriod = action.payload;
    },
    rejectEvaluationPeriod(state) {
      state.loadPeriodState = APIResponseStatus.Error;
    },
    submitEvaluationOrder(state) {
      state.submitOrderState = APIResponseStatus.InProgress;
    },
    rejectEvaluationOrder(state) {
      state.submitOrderState = APIResponseStatus.Error;
    },
    acceptEvaluationOrder(state, action: PayloadAction<string>) {
      state.submitOrderState = APIResponseStatus.Success;
      state.evaluationPaymentRedirect = action.payload;
    },
    setExaminationParts(
      state,
      action: PayloadAction<Partial<ExaminationParts>>,
    ) {
      state.examinationParts = { ...state.examinationParts, ...action.payload };
    },
    setAcceptConditions(state, action: PayloadAction<boolean>) {
      state.acceptConditions = action.payload;
    },
    setParticipantDetails(
      state,
      action: PayloadAction<Partial<ParticipantDetails>>,
    ) {
      state.participantDetails = {
        ...state.participantDetails,
        ...action.payload,
      };
    },
    setShowErrors(state, action: PayloadAction<boolean>) {
      state.showErrors = action.payload;
    },
    resetEvaluationOrderState(_state) {
      return initialState;
    },
    acceptEvaluationOrderDetails(
      state,
      action: PayloadAction<EvaluationOrderDetails>,
    ) {
      state.loadEvaluationOrderDetailsState = APIResponseStatus.Success;
      state.evaluationOrderDetails = action.payload;
    },
    loadEvaluationOrderDetails(state, _action: PayloadAction<number>) {
      state.loadEvaluationOrderDetailsState = APIResponseStatus.InProgress;
    },
    rejectEvaluationOrderDetails(state) {
      state.loadEvaluationOrderDetailsState = APIResponseStatus.Error;
    },
  },
});

export const evaluationOrderReducer = evaluationOrderSlice.reducer;
export const {
  loadEvaluationPeriod,
  storeEvaluationPeriod,
  rejectEvaluationPeriod,
  rejectEvaluationOrder,
  setAcceptConditions,
  setExaminationParts,
  setParticipantDetails,
  setShowErrors,
  submitEvaluationOrder,
  resetEvaluationOrderState,
  acceptEvaluationOrder,
  acceptEvaluationOrderDetails,
  loadEvaluationOrderDetails,
  rejectEvaluationOrderDetails,
} = evaluationOrderSlice.actions;
