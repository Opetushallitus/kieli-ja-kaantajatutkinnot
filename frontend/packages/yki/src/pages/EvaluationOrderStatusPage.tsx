import { Box } from '@mui/system';
import { useSearchParams } from 'react-router-dom';
import { H1, H2 } from 'shared/components';

import { usePublicTranslation } from 'configs/i18n';

enum OrderStatus {
  Success = 'payment-success',
  Cancel = 'payment-cancel',
  Error = 'payment-error',
}

const Success = () => {
  const { t } = usePublicTranslation({
    keyPrefix: 'yki.pages.evaluationOrderStatusPage.success',
  });

  return (
    <>
      <H1>{t('title')}</H1>
      <H2>{t('info')}</H2>
    </>
  );
};

const Cancel = () => {
  const { t } = usePublicTranslation({
    keyPrefix: 'yki.pages.evaluationOrderStatusPage.cancel',
  });

  return (
    <>
      <H1>{t('title')}</H1>
      <H2>{t('info')}</H2>
    </>
  );
};

const Error = () => {
  const { t } = usePublicTranslation({
    keyPrefix: 'yki.pages.evaluationOrderStatusPage.error',
  });

  return (
    <>
      <H1>{t('title')}</H1>
      <H2>{t('info')}</H2>
    </>
  );
};

const useOrderStatusComponent = () => {
  const [params] = useSearchParams();
  const status = params.get('status');
  switch (status) {
    case OrderStatus.Success:
      return <Success />;
    case OrderStatus.Cancel:
      return <Cancel />;
    default:
      return <Error />;
  }
};

export const EvaluationOrderStatusPage = () => {
  const orderStatus = useOrderStatusComponent();

  return <Box>{orderStatus}</Box>;
};
