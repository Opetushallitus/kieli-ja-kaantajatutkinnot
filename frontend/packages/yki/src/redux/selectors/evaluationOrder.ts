import { RootState } from 'configs/redux';

export const evaluationOrderSelector = (state: RootState) =>
  state.evaluationOrder;
