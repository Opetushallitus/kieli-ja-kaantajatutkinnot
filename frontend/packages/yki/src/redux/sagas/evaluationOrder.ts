import { PayloadAction } from '@reduxjs/toolkit';
import { AxiosResponse } from 'axios';
import { call, put, select, takeLatest } from 'redux-saga/effects';

import axiosInstance from 'configs/axios';
import { getCurrentLang, translateOutsideComponent } from 'configs/i18n';
import { APIEndpoints } from 'enums/api';
import {
  EvaluationOrderDetailsResponse,
  EvaluationOrderResponse,
} from 'interfaces/evaluationOrder';
import { setAPIError } from 'redux/reducers/APIError';
import {
  acceptEvaluationOrder,
  acceptEvaluationOrderDetails,
  EvaluationOrderState,
  loadEvaluationOrderDetails,
  rejectEvaluationOrder,
  rejectEvaluationOrderDetails,
  submitEvaluationOrder,
} from 'redux/reducers/evaluationOrder';
import { evaluationOrderSelector } from 'redux/selectors/evaluationOrder';
import { SerializationUtils } from 'utils/serialization';

function* submitEvaluationOrderSaga() {
  const t = translateOutsideComponent();
  try {
    const evaluationOrder: EvaluationOrderState = yield select(
      evaluationOrderSelector,
    );
    const lang = getCurrentLang();
    const response: AxiosResponse<EvaluationOrderResponse> = yield call(
      axiosInstance.post,
      APIEndpoints.EvaluationOrder.replace(
        /:evaluationId/,
        `${evaluationOrder.evaluationPeriod?.id}`,
      ),
      JSON.stringify(
        SerializationUtils.serializeEvaluationOrder(evaluationOrder),
      ),
      {
        params: {
          lang: SerializationUtils.serializeAppLanguage(lang),
          'use-yki-ui': true,
        },
      },
    );

    yield put(acceptEvaluationOrder(response.data.redirect));
  } catch (error) {
    yield put(rejectEvaluationOrder());
    yield put(setAPIError(t('yki.common.error')));
  }
}

function* loadEvaluationOrderDetailsSaga(action: PayloadAction<number>) {
  try {
    const lang = getCurrentLang();
    const id = action.payload;
    const response: AxiosResponse<EvaluationOrderDetailsResponse> = yield call(
      axiosInstance.get,
      APIEndpoints.EvaluationOrderDetails.replace(
        /:evaluationOrderId/,
        `${id}`,
      ),
      { params: { lang: SerializationUtils.serializeAppLanguage(lang) } },
    );
    yield put(
      acceptEvaluationOrderDetails(
        SerializationUtils.deserializeEvaluationOrderDetailsResponse(
          response.data,
        ),
      ),
    );
  } catch (error) {
    yield put(rejectEvaluationOrderDetails());
  }
}

export function* watchEvaluationOrder() {
  yield takeLatest(submitEvaluationOrder.type, submitEvaluationOrderSaga);
  yield takeLatest(
    loadEvaluationOrderDetails.type,
    loadEvaluationOrderDetailsSaga,
  );
}
