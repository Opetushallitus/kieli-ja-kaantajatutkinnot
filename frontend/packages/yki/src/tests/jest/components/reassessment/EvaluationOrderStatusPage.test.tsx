import { PreloadedState } from '@reduxjs/toolkit';
import * as reactRouterDom from 'react-router-dom';
import renderer from 'react-test-renderer';
import { APIResponseStatus } from 'shared/enums';

import { initI18nForTests } from 'configs/i18n';
import { RootState } from 'configs/redux';
import { PaymentStatus } from 'enums/api';
import EvaluationOrderStatusPage from 'pages/EvaluationOrderStatusPage';
import { initialState as initialEvaluationOrderState } from 'redux/reducers/evaluationOrder';
import { DefaultProviders } from 'tests/jest/utils/DefaultProviders';
import { evaluationOrderDetailsResponse } from 'tests/msw/fixtures/evaluationOrder';
import { SerializationUtils } from 'utils/serialization';

jest.unmock('configs/i18n');
jest.requireActual('configs/i18n');

beforeAll(() => {
  initI18nForTests();
});

const mockUseSearchParams = (params: URLSearchParams) => {
  jest
    .spyOn(reactRouterDom, 'useSearchParams')
    .mockReturnValue([params, jest.fn()]);
};

const preloadedState: PreloadedState<RootState> = {
  evaluationOrder: {
    ...initialEvaluationOrderState,
    loadEvaluationOrderDetailsState: APIResponseStatus.Success,
    evaluationOrderDetails:
      SerializationUtils.deserializeEvaluationOrderDetailsResponse(
        evaluationOrderDetailsResponse
      ),
  },
};

describe('EvaluationOrderStatusPage', () => {
  it('should render view correctly on successful payment', () => {
    mockUseSearchParams(
      new URLSearchParams({ id: '1', status: PaymentStatus.Success })
    );
    const tree = renderer
      .create(
        <DefaultProviders preloadedState={preloadedState}>
          <EvaluationOrderStatusPage />
        </DefaultProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render view correctly on cancelled payment', () => {
    mockUseSearchParams(
      new URLSearchParams({ id: '1', status: PaymentStatus.Cancel })
    );
    const tree = renderer
      .create(
        <DefaultProviders preloadedState={preloadedState}>
          <EvaluationOrderStatusPage />
        </DefaultProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render view correctly on payment error', () => {
    mockUseSearchParams(
      new URLSearchParams({ id: '1', status: PaymentStatus.Error })
    );
    const tree = renderer
      .create(
        <DefaultProviders preloadedState={preloadedState}>
          <EvaluationOrderStatusPage />
        </DefaultProviders>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
