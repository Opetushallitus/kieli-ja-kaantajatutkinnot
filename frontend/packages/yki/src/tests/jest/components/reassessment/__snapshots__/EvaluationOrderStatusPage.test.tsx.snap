// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvaluationOrderStatusPage should render view correctly on cancelled payment 1`] = `
<div
  className="public-evaluation-order-status-page MuiBox-root css-0"
>
  <div
    className="public-evaluation-order-status-page__heading"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Maksu keskeytyi
    </h1>
    <div
      className="header-separator MuiBox-root css-0"
    />
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 public-evaluation-order-status-page__paper rows gapped css-3su884-MuiPaper-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Voit yrittää maksua uudestaan. Käsittelemme vain maksetut tarkistusarviointipyynnöt.
    </p>
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium fit-content-max-width css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      href="/yki/ilmoittautuminen"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Takaisin aloitussivulle
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </a>
  </div>
</div>
`;

exports[`EvaluationOrderStatusPage should render view correctly on payment error 1`] = `
<div
  className="public-evaluation-order-status-page MuiBox-root css-0"
>
  <div
    className="public-evaluation-order-status-page__heading"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Maksu epäonnistui
    </h1>
    <div
      className="header-separator MuiBox-root css-0"
    />
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 public-evaluation-order-status-page__paper rows gapped css-3su884-MuiPaper-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Tarkistusarviointimaksun maksaminen ei onnistunut. Yritä maksua uudestaan.
    </p>
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium fit-content-max-width css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      href="/yki/ilmoittautuminen"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Takaisin aloitussivulle
    </a>
  </div>
</div>
`;

exports[`EvaluationOrderStatusPage should render view correctly on successful payment 1`] = `
<div
  className="public-evaluation-order-status-page MuiBox-root css-0"
>
  <div
    className="public-evaluation-order-status-page__heading"
  >
    <h1
      className="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
    >
      Olet pyytänyt tarkistusarviointia!
    </h1>
    <div
      className="header-separator MuiBox-root css-0"
    />
  </div>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 public-evaluation-order-status-page__paper rows gapped css-3su884-MuiPaper-root"
  >
    <div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Tutkinto
        :
         
        <b>
          suomi, keskitaso
        </b>
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Testipäivä
        :
         
        <b>
          10/11/2023
        </b>
      </p>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Sinulle on lähetetty vahvistus tarkistusarviointipyynnöstäsi sähköpostilla. Sähköpostissa on myös kuitti tarkistusarviointimaksustasi. Jos et ole saanut viestiä, tarkista roskapostikansiosi.
    </p>
    <a
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium fit-content-max-width css-zcbmsk-MuiButtonBase-root-MuiButton-root"
      href="/yki/ilmoittautuminen"
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      Takaisin aloitussivulle
    </a>
  </div>
</div>
`;
